### ISSUE TITLE
Extension cluster page visibility API should be finer grained

### ISSUE THREAD

**Issue by stefcameron** [kontena/lens#4591]
@jakolehm 
> IMHO this method currently passes wrong (legacy) object. It should actually give `KubernetesCluster`, not `Cluster` (which is "private" api).

There are actually 2 problems with this old API:
1. The one you mention.
2. The fact that it's not possible for an extension to have multiple cluster pages, and decide which one is visible and which one isn't -- it's "all or nothing". This is because Lens' visibility check is tied to the cluster, not the cluster page menu item.

So this is a good opportunity to fix this API:
1. Deprecate this API entirely. Using `public async isEnabledForCluster(cluster)` should result in a console warning, if possible. Might not be since people are encouraged to override it and not call its super impl. At least mark it deprecated in the docs.
2. Add a new option to check this in the cluster page menu item itself, so that the check is per menu item, given the activated cluster:

    ```javascript
    this.clusterPageMenus = [
      {
        target: { pageId: "page-id" },
        title: "My cluster page",
        isEnabled(entity) {
          return entity.metadata.source === "my-source";
        },
        components: {
          Icon: () => <MyIcon />,
        },
      },
    ];
    ```

    `isEnabled` should obviously default to `true` when not specified, and it's given the newly-activated entity instead of that old cluster object thing.

_Originally posted by @stefcameron in https://github.com/lensapp/lens/issues/3790#issuecomment-919491361_

**Comment by stefcameron** [comment]
Note that #4584 did not address the proposed API changes in #3790 so this issue is for that improvement.

**Comment by Nokel81** [comment]
This has been resolved by #5800, you can check against the current cluster by reading `Renderer.Catalog.activeCluster`
